core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/ChangeHandler.java

```diff
@@ -155,40 +136,40 @@ public static void handleBufferedChanges() {
   }
 
   /*package-private*/
-  static void addListeners(Releasable element) {
-    element.addReleaseListener(releaseListener);
-    if (element instanceof Element) {
-      ((Element) element).addPropertyListener(propertyListener);
-      if (element instanceof Component) {
-        ((Component) element).addAbsoluteTransformationListener(absoluteTransformationListener);
-        ((Component) element).addHierarchyListener(hierarchyListener);
-        if (element instanceof Composite) {
-          ((Composite) element).addChildrenListener(componentsListener);
+  static void addListeners(Releasable releasable) {
+    releasable.addReleaseListener(releaseListener);
+    if (releasable instanceof Element element) {
+      element.addPropertyListener(propertyListener);
+      if (releasable instanceof Component component) {
```

Comment:
would it be clearer if you called instanceof on the element variable here on the nested ifs? (this is not rhetorical, i can't quite decide)

Reply:
These listeners are questionable on many layers.
- They a globals.
- They are shared across all Releasables
- They all call the same method, `ChangeHandler::handleOrBufferEvent` that calls `handleEvent` which has another mass of chained `if instanceof` to differentiate.
I will save it for another day.